corehq/form_processor/backends/sql/update_strategy.py

```diff
@@ -287,6 +301,32 @@ def rebuild_from_transactions(self, transactions, rebuild_transaction, unarchive
         if not self.case.modified_on:
             self.case.modified_on = rebuild_transaction.server_date
 
+    def reconcile_transactions_if_necessary(self, xform):
+        if self.case.check_transaction_order():
+            return
+        try:
+            self.reconcile_transactions()
+        except ReconciliationError as e:
+            reconciliation_soft_assert(False, e.message)
+
+    def reconcile_transactions(self):
+        transactions = self.case.transactions
+        sorted_transactions = sorted(
+            transactions,
+            key=_transaction_sort_key_function(self.case)
+        )
```

Comment:
I think we should apply this same sorting to any other case rebuilds so I think we can move this into `rebuild_from_transactions` function.